src/codeu/chat/client/core/Context.java

```diff
@@ -41,6 +41,12 @@ public Context(ConnectionSource source) {
   }
 
   public UserContext create(String name) {
+	// check if user with that name already exists
+	for(User u: view.getUsers()) {
```

Comment:
Ya if that were the case, would probably have to send something to the server and have the model return the user since it can query by id.